daemon/algod/api/server/v2/handlers.go

```diff
@@ -1153,41 +1153,30 @@ func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64,
 	castedMax := nilToZero(params.Max)
 	maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication
 
-	// compute `maxReturnNum` from `castedMax` and `maxBoxThreshold := v2.Node.Config().MaxAPIBoxPerApplication`
-	// `maxReturnNum == 0` for returning everything
-	// - `castedMax == 0` and `maxBoxThreshold == 0`,
-	//       return everything
-	// - `castedMax == 0` and `maxBoxThreshold > 0`,
-	//       returns up to `maxBoxThreshold`, check if exceeds
-	// - `castedMax > 0` and `maxBoxThreshold == 0`,
-	//       return up to `castedMax`
-	// - `castedMax > 0` and `maxBoxThreshold > 0`,
-	//       return up to min(castedMax, maxBoxThreshold),
-	//       if return size is `maxBoxThreshold`, check if exceeds
-
-	var boxKeys []string
-	var err error
-
-	dominatedByQryParams := castedMax > 0 && (maxBoxThreshold >= castedMax || maxBoxThreshold == 0)
-	returnsAll := castedMax == 0 && maxBoxThreshold == 0
-
-	if dominatedByQryParams || returnsAll {
-		boxKeys, err = ledger.LookupKeysByPrefix(lastRound, keyPrefix, castedMax)
-		if err != nil {
-			return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
-		}
-	} else {
-		boxKeys, err = ledger.LookupKeysByPrefix(lastRound, keyPrefix, maxBoxThreshold+1)
-		if err != nil {
-			return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
-		}
-		if uint64(len(boxKeys)) > maxBoxThreshold {
-			v2.Log.Info("MaxAPIBoxPerApplication limit %d exceeded", maxBoxThreshold)
-			return ctx.JSON(http.StatusBadRequest, generated.ErrorResponse{
-				Message: "Result limit exceeded",
-				Data:    nil,
-			})
+	algodSupportsUnlimitedResults := maxBoxThreshold == 0
```

Comment:
Suppose for example `maxBoxThreshold == 1` and `castedMax == 0`.

Following this logic through, we get `maxKeys() == 2`. 

Why shouldn't we get `maxKeys() == 1` ? After all , that's what `maxBoxThreshold` seems to indicate.

Reply:
I get it... resolving